jsontableschema/types.py

```diff
@@ -73,16 +73,16 @@ def _check_constraint_list(self, value, constraint_list):
     def cast(self, value, skip_constraints=False):
         """Return boolean if `value` can be cast as type `self.py`."""
 
-        # We check on `constraints.required` before we cast
-        if not skip_constraints:
-            required = self._get_constraint_value('required')
-            if required is not None:
-                if not required and (value in (None, utilities.NULL_VALUES)):
-                    return None
-                elif required and value in (None, ''):
-                    raise exceptions.ConstraintError(
-                        msg="The field '{0}' requires a value".format(
-                            self.field_name))
+        # Return None/raise constraint error if value is NULL_VALUE
+        if value in ([None] + utilities.NULL_VALUES):
+            if not skip_constraints:
```

Comment:
Okay, discussed with @pwalsh.
Here's the thing:
In the current implementation, there's no way to load an empty string to the DB - either it's required, and we throw an error, or it's not required and we return None.
It's probably better to disallow empty strings by using the minLength constraint and allow empty strings as a valid value for string fields.